src/main/java/org/violetmoon/quark/content/experimental/module/NarratorReadoutModule.java

```diff
@@ -114,7 +114,21 @@ private String getReadout(Minecraft mc, boolean full) {
 					if(state.getBlock() instanceof SignBlock) {
 						SignBlockEntity tile = (SignBlockEntity) mc.level.getBlockEntity(pos);
 						sb.append(I18n.get("quark.readout.sign_says"));
-						for(Component cmp : tile.messages) {
+						for(Component cmp : tile.getFrontText().getMessages(false)) {
+							String msg = cmp.getString().trim();
+							if(!msg.isEmpty()) {
+								sb.append(cmp.getString());
+								sb.append(" ");
+							}
+						}
+
+						sb.append(". ");
+
+						//todo: Can someone check if this would be a decent way to handle back parts of sign? Alternatively, perhaps we could see if the raycast gets the front or back of the sign?
+						// No matter what, it should have a separate piece of text for front and back but still.
+
+						sb.append(I18n.get("quark.readout.sign_says"));
```

Comment:
Could someone look at this? I was wondering if this was the right way to go through with this but Im unsure.